#Importing the raw file:

---



In [1]:
import pandas as pd
df=pd.read_csv("https://raw.githubusercontent.com/SanchitKS12/data/refs/heads/master/weather-nominal-weka.csv")
df

outlook temperature humidity  windy play
0      sunny         hot     high  False   no
1      sunny         hot     high   True   no
2   overcast         hot     high  False  yes
3      rainy        mild     high  False  yes
4      rainy        cool   normal  False  yes
5      rainy        cool   normal   True   no
6   overcast        cool   normal   True  yes
7      sunny        mild     high  False   no
8      sunny        cool   normal  False  yes
9      rainy        mild   normal  False  yes
10     sunny        mild   normal   True  yes
11  overcast        mild     high   True  yes
12  overcast         hot   normal  False  yes
13     rainy        mild     high   True   no

#Getting info upon data:

In [2]:
df.describe()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14 entries, 0 to 13
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   outlook      14 non-null     object
 1   temperature  14 non-null     object
 2   humidity     14 non-null     object
 3   windy        14 non-null     bool  
 4   play         14 non-null     object
dtypes: bool(1), object(4)
memory usage: 594.0+ bytes


In [3]:
df.index=df.index+1
df

outlook temperature humidity  windy play
1      sunny         hot     high  False   no
2      sunny         hot     high   True   no
3   overcast         hot     high  False  yes
4      rainy        mild     high  False  yes
5      rainy        cool   normal  False  yes
6      rainy        cool   normal   True   no
7   overcast        cool   normal   True  yes
8      sunny        mild     high  False   no
9      sunny        cool   normal  False  yes
10     rainy        mild   normal  False  yes
11     sunny        mild   normal   True  yes
12  overcast        mild     high   True  yes
13  overcast         hot   normal  False  yes
14     rainy        mild     high   True   no

#Checking for discripancy in data:

In [4]:
df.isna()

outlook  temperature  humidity  windy   play
1     False        False     False  False  False
2     False        False     False  False  False
3     False        False     False  False  False
4     False        False     False  False  False
5     False        False     False  False  False
6     False        False     False  False  False
7     False        False     False  False  False
8     False        False     False  False  False
9     False        False     False  False  False
10    False        False     False  False  False
11    False        False     False  False  False
12    False        False     False  False  False
13    False        False     False  False  False
14    False        False     False  False  False

#Assigning,filtering,appointing the data as per need:

In [5]:
df = df.drop('play', axis=1, errors='ignore')
df

outlook temperature humidity  windy
1      sunny         hot     high  False
2      sunny         hot     high   True
3   overcast         hot     high  False
4      rainy        mild     high  False
5      rainy        cool   normal  False
6      rainy        cool   normal   True
7   overcast        cool   normal   True
8      sunny        mild     high  False
9      sunny        cool   normal  False
10     rainy        mild   normal  False
11     sunny        mild   normal   True
12  overcast        mild     high   True
13  overcast         hot   normal  False
14     rainy        mild     high   True

In [6]:
df.rename(columns={'outlook':'Weather'},inplace=True)
df.rename(columns={'temperature':'temperature(Celcius)'},inplace=True)
df.rename(columns={'humidity':'humidity(in %)'},inplace=True)
df

Weather temperature(Celcius) humidity(in %)  windy
1      sunny                  hot           high  False
2      sunny                  hot           high   True
3   overcast                  hot           high  False
4      rainy                 mild           high  False
5      rainy                 cool         normal  False
6      rainy                 cool         normal   True
7   overcast                 cool         normal   True
8      sunny                 mild           high  False
9      sunny                 cool         normal  False
10     rainy                 mild         normal  False
11     sunny                 mild         normal   True
12  overcast                 mild           high   True
13  overcast                  hot         normal  False
14     rainy                 mild           high   True

##For weather:

Hot: Around 30°C to 40°C or higher.
(Example: A sunny summer day in a desert region.)

Mild: Around 15°C to 25°C.
(Example: A pleasant spring or autumn day, neither too warm nor too cold.)

Cool: Around 10°C to 15°C.
(Example: A crisp autumn morning or early spring day where you might need a light jacket.)

In [7]:
def oye(a):
  if(a=='hot'):
    return 35
  if(a=='mild'):
    return 25
  if(a=='cool'):
    return 15
df['temperature(Celcius)']=df['temperature(Celcius)'].apply(oye)
df

Weather  temperature(Celcius) humidity(in %)  windy
1      sunny                    35           high  False
2      sunny                    35           high   True
3   overcast                    35           high  False
4      rainy                    25           high  False
5      rainy                    15         normal  False
6      rainy                    15         normal   True
7   overcast                    15         normal   True
8      sunny                    25           high  False
9      sunny                    15         normal  False
10     rainy                    25         normal  False
11     sunny                    25         normal   True
12  overcast                    25           high   True
13  overcast                    35         normal  False
14     rainy                    25           high   True

##For Humidity:

High humidity:

Above 70% relative humidity is generally considered high.
(Example: A humid, sticky day that feels uncomfortable and muggy.)

For weather (outdoor air):

Normal humidity is around 30% to 60% relative humidity.
(This range is comfortable and doesn't feel too dry or too humid.)

In [9]:
def hum(c):
  if(c=='high'):
    return 70
  if(c=='normal'):
    return 45
df['humidity(in %)']=df['humidity(in %)'].apply(hum)
df

Weather  temperature(Celcius)  humidity(in %)  windy
1      sunny                    35              70  False
2      sunny                    35              70   True
3   overcast                    35              70  False
4      rainy                    25              70  False
5      rainy                    15              45  False
6      rainy                    15              45   True
7   overcast                    15              45   True
8      sunny                    25              70  False
9      sunny                    15              45  False
10     rainy                    25              45  False
11     sunny                    25              45   True
12  overcast                    25              70   True
13  overcast                    35              45  False
14     rainy                    25              70   True

In [12]:
df['temperature(Celcius)'].astype(float)
df['humidity(in %)'].astype(float)
df

Weather  temperature(Celcius)  humidity(in %)  windy
1      sunny                    35              70  False
2      sunny                    35              70   True
3   overcast                    35              70  False
4      rainy                    25              70  False
5      rainy                    15              45  False
6      rainy                    15              45   True
7   overcast                    15              45   True
8      sunny                    25              70  False
9      sunny                    15              45  False
10     rainy                    25              45  False
11     sunny                    25              45   True
12  overcast                    25              70   True
13  overcast                    35              45  False
14     rainy                    25              70   True

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14 entries, 1 to 14
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Weather               14 non-null     object
 1   temperature(Celcius)  14 non-null     int64 
 2   humidity(in %)        14 non-null     int64 
 3   windy                 14 non-null     bool  
dtypes: bool(1), int64(2), object(1)
memory usage: 482.0+ bytes


## Main Anyalysis:

In [14]:
week_1=df.loc[1:7]
week_1

Weather  temperature(Celcius)  humidity(in %)  windy
1     sunny                    35              70  False
2     sunny                    35              70   True
3  overcast                    35              70  False
4     rainy                    25              70  False
5     rainy                    15              45  False
6     rainy                    15              45   True
7  overcast                    15              45   True

In [15]:
week_2=df.loc[8:14]
week_2

Weather  temperature(Celcius)  humidity(in %)  windy
8      sunny                    25              70  False
9      sunny                    15              45  False
10     rainy                    25              45  False
11     sunny                    25              45   True
12  overcast                    25              70   True
13  overcast                    35              45  False
14     rainy                    25              70   True

In [16]:
week_1['Weather'].value_counts()

Weather
rainy       3
sunny       2
overcast    2
Name: count, dtype: int64

In [17]:
week_2['Weather'].value_counts()

Weather
sunny       3
rainy       2
overcast    2
Name: count, dtype: int64

In [22]:
p=week_1.sample(4,random_state=1)
p

Weather  temperature(Celcius)  humidity(in %)  windy
7  overcast                    15              45   True
3  overcast                    35              70  False
2     sunny                    35              70   True
1     sunny                    35              70  False

In [23]:
q=week_2.sample(4,random_state=1)
q

Weather  temperature(Celcius)  humidity(in %)  windy
14   rainy                    25              70   True
10   rainy                    25              45  False
9    sunny                    15              45  False
8    sunny                    25              70  False

In [24]:
p['temperature(Celcius)'].value_counts()

temperature(Celcius)
35    3
15    1
Name: count, dtype: int64

In [26]:
q['temperature(Celcius)'].value_counts()

temperature(Celcius)
25    3
15    1
Name: count, dtype: int64

In [27]:
print(p['temperature(Celcius)'].mean())
print(q['temperature(Celcius)'].mean())

30.0
22.5


In [28]:
print((p['humidity(in %)'].cumsum())/4)
print((q['humidity(in %)'].cumsum())/4) #Not necessary but might come in handy

7    11.25
3    28.75
2    46.25
1    63.75
Name: humidity(in %), dtype: float64
14    17.50
10    28.75
9     40.00
8     57.50
Name: humidity(in %), dtype: float64


In [30]:
print((p['humidity(in %)'].mean()))
print((q['humidity(in %)'].mean()))

63.75
57.5


#Conclusion:

Based on all the data cleansing,filtering,appointing,etc. i can safely come to the conclusion that the upcoming week will have mostly sunny days and a relatively  humid environment with an average of 60% humidity and temperature ranging between 22.5 -30 degree Celcius.